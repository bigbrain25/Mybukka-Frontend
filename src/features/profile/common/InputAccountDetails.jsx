import React from 'react';

import PropTypes from 'prop-types';

import InputField from 'Components/input/InputField';

import EditTogglerButton from './EditTogglerButton';

import './inputAccountDetails.scss';

const InputAccountDetails = ({
  placeHolder,
  name,
  type,
  defaultValue,
  children,
}) => (
  <div className="input-acc-details-section">
    <form className="form-acc-details">
      <div className="form-group input-acc-details">
        {children || (
          <InputField
            type={type}
            placeholderText={placeHolder}
            classNames="input-acc-details"
            name={name}
            handleChange={() => {}}
            defaultValue={defaultValue}
          />
        )}
      </div>
      <EditTogglerButton text="EDIT" />
    </form>
  </div>
);

export default InputAccountDetails;

InputAccountDetails.defaultProps = {
  type: 'text',
  defaultValue: '',
  children: '',
};

InputAccountDetails.propTypes = {
  placeHolder: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  type: PropTypes.string,
  defaultValue: PropTypes.string,
  children: PropTypes.node,
};
