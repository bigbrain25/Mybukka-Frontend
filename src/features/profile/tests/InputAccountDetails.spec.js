import React from 'react';

import InputAccountDetails from '../common/InputAccountDetails';

describe('InputAccountDetails component', () => {
  const props = {
    placeHolder: 'First Name',
    name: 'firstName',
    type: 'firstName',
    defaultValue: 'Value',
  };

  const wrapper = mount(
    <InputAccountDetails {...props}><p>child</p></InputAccountDetails>
  );

  it('renders properly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
