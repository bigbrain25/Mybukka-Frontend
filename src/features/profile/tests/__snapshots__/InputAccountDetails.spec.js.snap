// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputAccountDetails component renders properly 1`] = `
ReactWrapper {
  Symbol(enzyme.__unrendered__): <InputAccountDetails
    defaultValue="Value"
    name="firstName"
    placeHolder="First Name"
    type="firstName"
  >
    <p>
      child
    </p>
  </InputAccountDetails>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <p>
        child
      </p>,
      "defaultValue": "Value",
      "name": "firstName",
      "placeHolder": "First Name",
      "type": "firstName",
    },
    "ref": null,
    "rendered": Object {
      "instance": <div
        class="input-acc-details-section"
      >
        <form
          class="form-acc-details"
        >
          <div
            class="form-group input-acc-details"
          >
            <p>
              child
            </p>
          </div>
          <div
            class="edit-toggler-button"
          >
            <button
              class="edit-save-toggler d-flex justify-content-end"
              data-target=""
              data-testid="button"
              data-toggle=""
              id="PNnYoMYuX"
              type="button"
            >
              EDIT
            </button>
          </div>
        </form>
      </div>,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form
          className="form-acc-details"
        >
          <div
            className="form-group input-acc-details"
          >
            <p>
              child
            </p>
          </div>
          <EditTogglerButton
            text="EDIT"
          />
        </form>,
        "className": "input-acc-details-section",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": <form
            class="form-acc-details"
          >
            <div
              class="form-group input-acc-details"
            >
              <p>
                child
              </p>
            </div>
            <div
              class="edit-toggler-button"
            >
              <button
                class="edit-save-toggler d-flex justify-content-end"
                data-target=""
                data-testid="button"
                data-toggle=""
                id="PNnYoMYuX"
                type="button"
              >
                EDIT
              </button>
            </div>
          </form>,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="form-group input-acc-details"
              >
                <p>
                  child
                </p>
              </div>,
              <EditTogglerButton
                text="EDIT"
              />,
            ],
            "className": "form-acc-details",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": <div
                class="form-group input-acc-details"
              >
                <p>
                  child
                </p>
              </div>,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <p>
                  child
                </p>,
                "className": "form-group input-acc-details",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": <p>
                    child
                  </p>,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "child",
                  },
                  "ref": null,
                  "rendered": Array [
                    "child",
                  ],
                  "type": "p",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "text": "EDIT",
              },
              "ref": null,
              "rendered": Object {
                "instance": <div
                  class="edit-toggler-button"
                >
                  <button
                    class="edit-save-toggler d-flex justify-content-end"
                    data-target=""
                    data-testid="button"
                    data-toggle=""
                    id="PNnYoMYuX"
                    type="button"
                  >
                    EDIT
                  </button>
                </div>,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <Button
                    classNames="edit-save-toggler d-flex justify-content-end"
                    dataTarget=""
                    dataToggle=""
                    disabled={false}
                    handleClick={[Function]}
                    id="PNnYoMYuX"
                    text="EDIT"
                    type="button"
                  >
                    <div />
                  </Button>,
                  "className": "edit-toggler-button",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "function",
                    "props": Object {
                      "children": <div />,
                      "classNames": "edit-save-toggler d-flex justify-content-end",
                      "dataTarget": "",
                      "dataToggle": "",
                      "disabled": false,
                      "handleClick": [Function],
                      "id": "PNnYoMYuX",
                      "text": "EDIT",
                      "type": "button",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": <button
                        class="edit-save-toggler d-flex justify-content-end"
                        data-target=""
                        data-testid="button"
                        data-toggle=""
                        id="PNnYoMYuX"
                        type="button"
                      >
                        EDIT
                      </button>,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "EDIT",
                        "className": "edit-save-toggler d-flex justify-content-end",
                        "data-target": "",
                        "data-testid": "button",
                        "data-toggle": "",
                        "disabled": false,
                        "id": "PNnYoMYuX",
                        "onClick": [Function],
                        "type": "button",
                      },
                      "ref": null,
                      "rendered": Array [
                        "EDIT",
                      ],
                      "type": "button",
                    },
                    "type": [Function],
                  },
                ],
                "type": "div",
              },
              "type": [Function],
            },
          ],
          "type": "form",
        },
      ],
      "type": "div",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <p>
          child
        </p>,
        "defaultValue": "Value",
        "name": "firstName",
        "placeHolder": "First Name",
        "type": "firstName",
      },
      "ref": null,
      "rendered": Object {
        "instance": <div
          class="input-acc-details-section"
        >
          <form
            class="form-acc-details"
          >
            <div
              class="form-group input-acc-details"
            >
              <p>
                child
              </p>
            </div>
            <div
              class="edit-toggler-button"
            >
              <button
                class="edit-save-toggler d-flex justify-content-end"
                data-target=""
                data-testid="button"
                data-toggle=""
                id="PNnYoMYuX"
                type="button"
              >
                EDIT
              </button>
            </div>
          </form>
        </div>,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <form
            className="form-acc-details"
          >
            <div
              className="form-group input-acc-details"
            >
              <p>
                child
              </p>
            </div>
            <EditTogglerButton
              text="EDIT"
            />
          </form>,
          "className": "input-acc-details-section",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": <form
              class="form-acc-details"
            >
              <div
                class="form-group input-acc-details"
              >
                <p>
                  child
                </p>
              </div>
              <div
                class="edit-toggler-button"
              >
                <button
                  class="edit-save-toggler d-flex justify-content-end"
                  data-target=""
                  data-testid="button"
                  data-toggle=""
                  id="PNnYoMYuX"
                  type="button"
                >
                  EDIT
                </button>
              </div>
            </form>,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="form-group input-acc-details"
                >
                  <p>
                    child
                  </p>
                </div>,
                <EditTogglerButton
                  text="EDIT"
                />,
              ],
              "className": "form-acc-details",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": <div
                  class="form-group input-acc-details"
                >
                  <p>
                    child
                  </p>
                </div>,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <p>
                    child
                  </p>,
                  "className": "form-group input-acc-details",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": <p>
                      child
                    </p>,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "child",
                    },
                    "ref": null,
                    "rendered": Array [
                      "child",
                    ],
                    "type": "p",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "text": "EDIT",
                },
                "ref": null,
                "rendered": Object {
                  "instance": <div
                    class="edit-toggler-button"
                  >
                    <button
                      class="edit-save-toggler d-flex justify-content-end"
                      data-target=""
                      data-testid="button"
                      data-toggle=""
                      id="PNnYoMYuX"
                      type="button"
                    >
                      EDIT
                    </button>
                  </div>,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <Button
                      classNames="edit-save-toggler d-flex justify-content-end"
                      dataTarget=""
                      dataToggle=""
                      disabled={false}
                      handleClick={[Function]}
                      id="PNnYoMYuX"
                      text="EDIT"
                      type="button"
                    >
                      <div />
                    </Button>,
                    "className": "edit-toggler-button",
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "function",
                      "props": Object {
                        "children": <div />,
                        "classNames": "edit-save-toggler d-flex justify-content-end",
                        "dataTarget": "",
                        "dataToggle": "",
                        "disabled": false,
                        "handleClick": [Function],
                        "id": "PNnYoMYuX",
                        "text": "EDIT",
                        "type": "button",
                      },
                      "ref": null,
                      "rendered": Object {
                        "instance": <button
                          class="edit-save-toggler d-flex justify-content-end"
                          data-target=""
                          data-testid="button"
                          data-toggle=""
                          id="PNnYoMYuX"
                          type="button"
                        >
                          EDIT
                        </button>,
                        "key": undefined,
                        "nodeType": "host",
                        "props": Object {
                          "children": "EDIT",
                          "className": "edit-save-toggler d-flex justify-content-end",
                          "data-target": "",
                          "data-testid": "button",
                          "data-toggle": "",
                          "disabled": false,
                          "id": "PNnYoMYuX",
                          "onClick": [Function],
                          "type": "button",
                        },
                        "ref": null,
                        "rendered": Array [
                          "EDIT",
                        ],
                        "type": "button",
                      },
                      "type": [Function],
                    },
                  ],
                  "type": "div",
                },
                "type": [Function],
              },
            ],
            "type": "form",
          },
        ],
        "type": "div",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
